api/v1alpha1/ext_proc_types.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+type ExtProcHeaderProcessingMode string
+
+const (
+	DefaultExtProcHeaderProcessingMode ExtProcHeaderProcessingMode = "DEFAULT"
+	SendExtProcHeaderProcessingMode    ExtProcHeaderProcessingMode = "SEND"
+	SkipExtProcHeaderProcessingMode    ExtProcHeaderProcessingMode = "SKIP"
+)
+
+type ExtProcBodyProcessingMode string
+
+const (
+	NoneExtProcHeaderProcessingMode            ExtProcBodyProcessingMode = "NONE"
+	StreamedExtProcHeaderProcessingMode        ExtProcBodyProcessingMode = "STREAMED"
+	BufferedExtProcHeaderProcessingMode        ExtProcBodyProcessingMode = "BUFFERED"
+	BufferedPartialExtProcHeaderProcessingMode ExtProcBodyProcessingMode = "BUFFERED_PARTIAL"
+)
+
+type ProcessingModeOptions struct {
+	// Defines header processing mode
+	Headers *ExtProcHeaderProcessingMode `json:"request,omitempty"`
+	// Defines body processing mode
+	Body *ExtProcBodyProcessingMode `json:"response,omitempty"`
+}
+
+// ExtProcProcessingMode defines if and how headers and bodies are sent to the service.
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_proc/v3/processing_mode.proto#envoy-v3-api-msg-extensions-filters-http-ext-proc-v3-processingmode
+type ExtProcProcessingMode struct {
+	// Defines header and body treatment for requests
+	// Default: SEND
```

Comment:
Set the default with 
```golang
// +kubebuilder:default:=Send
```

( https://book.kubebuilder.io/reference/markers/crd-validation )